webapp/channels/src/components/user_settings/notifications/user_settings_notifications.tsx

```diff
@@ -145,9 +148,10 @@ function getDefaultStateFromProps(props: Props): State {
     let channelKey = false;
     let isCustomKeysWithNotificationInputChecked = false;
     const customKeysWithNotification: MultiInputValue[] = [];
+    const customKeysWithHighlight: MultiInputValue[] = [];
 
     if (props.user.notify_props) {
-        if (props.user.notify_props.mention_keys) {
+        if (props.user.notify_props?.mention_keys?.length > 0) {
```

Comment:
I don't think these `?.` are needed since we null checked `props.user.notify_props` above

Reply:
Yeah once the app crashed on new user because the prop `highlight_keys` was never set for it, so i added these checks

Reply:
I think we needed to check for `highlight_keys`  specifically because it didn't previously exist in the `UserNotifyProps`, but I don't think we need to null check `notify_props` or `mention_keys`